src/spikegadgets_to_nwb/spike_gadgets_raw_io.py

```diff
@@ -537,6 +537,93 @@ def get_analogsignal_multiplexed(self, channel_names=None) -> np.ndarray:
+        i_stop: int,
+        channel_names: list = None,
+        padding: int = 30000,
+    ) -> np.ndarray:
+        """Alternative method to access part of the multiplexed data.
+        Not memory efficient for many calls because it reads a buffer chunk before the requested data.
+        Better than get_analogsignal_multiplexed when need one call to specific time region
+
+        Parameters
+        ----------
+        i_start : int
+            index start
+        i_stop : int
+            index stop
+        channel_names : list[str], optional
+            channels to get, by default None will get all multiplex channels
+        padding : int, optional
+            how many packets before the desired series to load to ensure every channel receives update before requested,
+            by default 30000
+
+        Returns
+        -------
+        np.ndarray
+            multiplex data
+
+        Raises
+        ------
+        ValueError
+            _description_
+        """
+        print("compute multiplex cache", self.filename)
+        if channel_names is None:
+            # read all multiplexed channels
+            channel_names = list(self.multiplexed_channel_xml.keys())
+        else:
+            for ch_name in channel_names:
+                if ch_name not in self.multiplexed_channel_xml:
+                    raise ValueError(f"Channel name '{ch_name}' not found in file.")
+        # determine which packets to get from data
+        padding = min(padding, i_start)
+        i_start = i_start - padding
+        if i_stop is None:
+            i_stop = self._raw_memmap.shape[0]
+
+        # Make object to hold data
+        num_packet = i_stop - i_start
+        analog_multiplexed_data = np.empty(
+            (num_packet, len(channel_names)), dtype=np.int16
+        )
+
+        # precompute the static data offsets
+        data_offsets = np.empty((len(channel_names), 3), dtype=int)
+        for j, ch_name in enumerate(channel_names):
+            ch_xml = self.multiplexed_channel_xml[ch_name]
+            data_offsets[j, 0] = int(
+                self._multiplexed_byte_start + int(ch_xml.attrib["startByte"])
+            )
+            data_offsets[j, 1] = int(ch_xml.attrib["interleavedDataIDByte"])
+            data_offsets[j, 2] = int(ch_xml.attrib["interleavedDataIDBit"])
+        interleaved_data_id_byte_values = self._raw_memmap[
+            i_start:i_stop, data_offsets[:, 1]
+        ]
+        interleaved_data_id_bit_values = (
+            interleaved_data_id_byte_values >> data_offsets[:, 2]
+        ) & 1
+        # calculate which packets encode for which channel
+        initialize_stream_mask = np.logical_or(
+            (np.arange(num_packet) == 0)[:, None], interleaved_data_id_bit_values == 1
+        )
+        # read the data into int16
+        data = (
+            self._raw_memmap[i_start:i_stop, data_offsets[:, 0]]
+            + self._raw_memmap[i_start:i_stop, data_offsets[:, 0] + 1] * 256
```

Comment:
Where does this 256 come from? int16?

Reply:
Yeah, this was a way to vectorize the conversion

Reply:
Can you make this an informative constant just so there are no magic variables.